src/agora/doer.h

```diff
@@ -60,6 +79,16 @@ class Doer {
     return {};
   }
 
+  size_t frame_id_ = 0;
```

Comment:
are all of these private variables necessary for the time keeping function?  Can you better encapsulate the time tracking variables and features?

Reply:
If they are necessary as a class member function, at least group them

Reply:
only this class part I could not work. this requires lot of changes. we can discuss.